ProcessTree.cpp

```diff
@@ -485,7 +524,32 @@ void ProcessTree::UpdateFlags() {
     for (auto p : _processes) {
         ApplyFlags(p.second);
     }
-} 
+}
+
+void ProcessTree::SetContainerId(std::shared_ptr<ProcessTreeItem> p, std::string containerid)
+{
+    for (auto c : p->_children) {
+        auto it2 = _processes.find(c);
+        if (it2 != _processes.end()) {
+            auto cp = it2->second;
+            cp->_containerid = containerid;
+            SetContainerId(cp, containerid);
+        }
+    }
+}
+
+std::string ProcessTree::GetContainerId(std::string exe, const std::string& cmdline)
+{
+    std::string containerid = "";
+    if (ends_with(exe, "/containerd-shim") && starts_with(cmdline, "containerd-shim -namespace moby")) {
+        std::regex rgx(_cmdline_containerid_match_re);
```

Comment:
Please consider doing explicit parsing (e.g. Find -workdir substring, get following arg, then get id that follows last '/' in arg.